mfs/dir.go

```diff
@@ -131,7 +131,7 @@ func (d *Directory) cacheNode(name string, nd *dag.Node) (FSNode, error) {
 		ndir := NewDirectory(d.ctx, name, nd, d, d.dserv)
 		d.childDirs[name] = ndir
 		return ndir, nil
-	case ufspb.Data_File, ufspb.Data_Raw:
+	case ufspb.Data_File, ufspb.Data_Raw, ufspb.Data_Symlink:
```

Comment:
If we pass a dag node that represents a symlink to 'new file' here, its not going to work properly if we try to use it (`NewDagModifier` will produce undefined behaviour when trying to modify a symlink).